d3b_utils/aws_bucket_contents.py

```diff
@@ -0,0 +1,81 @@
+import boto3
+import pandas as pd
+
+
+def fetch_aws_bucket_obj_info(bucket_name, path_to_files):
+    """
+    List the objects of a bucket and the size of each object. Returns a
+    dict. Inspiration from
+    https://alexwlchan.net/2019/07/listing-s3-keys/
+
+    :param bucket_name: The name of the bucket.
+    :param path_to_files: path to the specific folder in the given
+        bucket where the files you want to list are. Exclude leading
+        '/' but include a '/' at the end of the folder. Should look
```

Comment:
I'm thinking something like this:

```suggestion
# Drop leading / from search prefix
[re.sub(r"^/", "", prefix) for prefix in [search_prefixes]]
```

Should I use different regex? With this solution, would it be better to enclose `search_prefixes` in brackets like i did here, or do so when `search_prefixes = [search_prefixes] or [""]` like your above comment?

Reply:
You can't put it in brackets here, because if it's already a list then you've just made a list of lists. I don't have it in brackets in my example for the same reason.

Reply:
I'm thinking this then to treat everything as an iterable and remove those leading slashes

```suggestion
    if isinstance(search_prefixes, str):
        search_prefixes = [search_prefixes]

    # Drop leading non-word characters from prefixes
    search_prefixes = [re.sub(r"^/", "", prefix) for prefix in search_prefixes]
```